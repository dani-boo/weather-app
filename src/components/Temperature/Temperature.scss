.field input {
  border: 4px solid #080808;
  width: 3em;
  height: 2em;
}
