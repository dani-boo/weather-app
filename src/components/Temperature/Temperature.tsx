import React from 'react'
import './Temperature.scss'

export const filters = ['min', 'max']

export interface temperatureProps {
  getTemperatures: (e: React.SyntheticEvent<Element, Event>) => void
  handleNumberInput: (e: React.ChangeEvent<HTMLInputElement>) => void
  selectFilter: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export const refreshPage = () => {
  window.location.reload()
}

const Temperature = (props: temperatureProps) => {
  return (
    <>
      <h3>Search by temperature:</h3>
      <form onSubmit={props.getTemperatures}>
        <label className='field'>
          <input
            type='number'
            name='temperature'
            placeholder=''
            onChange={props.handleNumberInput}
          />
        </label>
        {filters?.map((filter) => (
          <label key={filter}>
            <input
              type='radio'
              name='filter'
              value={filter}
              onChange={props.selectFilter}
            />
            {filter}
          </label>
        ))}
        <button className='button' type='submit'>
          Filter results
        </button>
      </form>
      <button className='button' onClick={refreshPage}>
        Refresh page
      </button>
    </>
  )
}

export default Temperature
