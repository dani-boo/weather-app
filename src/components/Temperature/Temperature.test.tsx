import React from 'react'
import { shallow } from 'enzyme'
import Temperature, { refreshPage } from './Temperature'

describe('refreshPage', () => {
  it('should call window.location.reload', () => {
    jest.spyOn(window.location, 'reload').mockImplementationOnce(() => {})
    refreshPage()
    expect(window.location.reload).toHaveBeenCalled()
  })
})
describe('Temperature', () => {
  it('should render with props', () => {
    const minProps = {
      getTemperatures: jest.fn(() => {}),
      handleNumberInput: jest.fn(() => {}),
      selectFilter: jest.fn(() => {}),
    }
    const wrapper = shallow(<Temperature {...minProps} />)
    expect(wrapper.find('form').length).toBe(1)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
