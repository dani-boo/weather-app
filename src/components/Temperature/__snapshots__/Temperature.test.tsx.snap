// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature should render with props 1`] = `"<h3>Search by temperature:</h3><form><label class=\\"field\\"><input type=\\"number\\" name=\\"temperature\\" placeholder=\\"\\"/></label><label><input type=\\"radio\\" name=\\"filter\\" value=\\"min\\"/>min</label><label><input type=\\"radio\\" name=\\"filter\\" value=\\"max\\"/>max</label><button class=\\"button\\" type=\\"submit\\">Filter results</button></form><button class=\\"button\\">Refresh page</button>"`;
